gapic/samplegen/samplegen.py

```diff
@@ -280,14 +280,17 @@ def validate_and_transform_request(self,
                     # and requires less munging of the assigned value
                     duplicate["value"] = f"'{val}'"
                     break
+                elif attr.is_primitive:
+                    # Only valid if this is the last attribute in the chain.
+                    break
                 else:
                     raise TypeError
 
             if i != len(attr_chain) - 1:
-                # We broke out of the loop after processing an enum.
+                # We broke out of the loop after processing an enum or a primitive.
```

Comment:
I know the other PR to handle resource names is still going, so just a remind here: when the primitive is a string, a `%` can still come after it to specify resource name entities. (It might not be a concern at all depending on how you have implemented resource name construction :) )

Reply:
All this code is lifted up into a method that is called only if the field being set is _not_ a resource name. I figured that this logic was too complicated and involved to mix with resource names, so I just partitioned it off.